### DISCUSSION TITLE
Reach compile taking forever to run.

### DISCUSSION THREAD

**Discussion by BakungaBronson** [reach-sh/reach-lang#877] [Q&A] [answered]
Hello.

Earlier today I was able to run ./reach compile until I put my laptop to sleep and ever since then when I run ./reach compile it shows a blank terminal no matter how long I wait.

I have tried ./reach down, ./reach docker-reset and ./reach update. On top of that I have tried to run ./reach down and ./reach docker-reset after update but that hasn't worked. I also restarted my laptop just to make sure.

I'm running reach inside of WSL 2, anyone have any idea on how I can fix this? 

I've also been asked to attach the .rsh file so that's available as well however Github does not support the file format so I have uploaded it here: https://drive.google.com/file/d/1c7W2Qx9f2RHYWwa8LUg9hQ5Y-ElPOXQ9/view?usp=sharing

**Comment by BakungaBronson** [comment, marked as answer]
I noticed that Docker is also running pretty high on system resource usage, not sure if the two are related.
![image](https://user-images.githubusercontent.com/51344005/166089501-1468eef4-c6a9-479f-afe7-5fb22595e669.png)

**Reply by jeapostrophe** [reply]
You can run `docker ps -a` to see what Docker is doing (or use the Docker Desktop UI) and you can run `reach down` to kill all of Reach's containers

**Reply by jeapostrophe** [reply]
Okay, I ran your program and it does seem to run extremely long. I'll investigate and get back to you! Thank you!

**Reply by BakungaBronson** [reply]
Thank you too.

**Reply by jeapostrophe** [reply]
K, I believe this is "normal" because your program just has huge path explosion when we analyze it. If you wait long enough, it will return. I think there are ways we might be able to improve the compilation, but the source of problem is that you have 81 elementary array that we have analyze combinatorially.

**Reply by BakungaBronson** [reply]
Oh ok, noted. Which is weird because it was compiling just fine before. Thank you for this!  Any idea which line in particular might be causing this?

**Reply by jeapostrophe** [reply]
The root of the problem is that array references inhibit the optimizer because we have to prove that they are in bounds. If you bundle them all together it is fine: https://gist.github.com/jeapostrophe/fc77fd7f6cc045330ebcae90a7030f09

**Reply by jeapostrophe** [reply]
See `catTrapped`

**Reply by BakungaBronson** [reply]
Got it. I'll take that into consideration going forward. Thank you again.
